File: sdks/kotlin/src/main/kotlin/hooksniff/sdk/wrapper/SearchResource.kt
package hooksniff.sdk.wrapper

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

/**
 * Search deliveries and events.
 */
class SearchResource(private val client: HookSniff) {

    private val gson = Gson()
    private val mapType = object : TypeToken<Map<String, Any?>>() {}.type

    /** Search deliveries. */
    fun query(input: Map<String, Any?>): Map<String, Any?> {
        val json = client.post("/v1/search", gson.toJson(input))
        return gson.fromJson(json, mapType)
    }
}

File: sdks/swift/HookSniffSDK/Sources/HookSniff/SearchResource.swift
import Foundation

/// Search resource — search deliveries and events.
public final class SearchResource {
    private let client: HookSniff

    init(client: HookSniff) {
        self.client = client
    }

    /// Search deliveries and events.
    public func search(_ params: [String: Any]) async throws -> [String: Any] {
        let body = try await client.requestJSON(method: "POST", path: "/v1/search", body: params)
        return JSONHelpers.dict(from: body)
    }
}
